test/expm.jl

```diff
@@ -1,6 +1,7 @@
 using StaticArrays, Test, LinearAlgebra
 
 @testset "Matrix exponential" begin
+    @test exp(SMatrix{0,0,Int}())::SMatrix == SMatrix{0,0,Bool}()
```

Comment:
```suggestion
    @test exp(SMatrix{0,0,Int}())::SMatrix === SMatrix{0,0,Float64}()
```
This would also test type promotion, right?

Reply:
Yes it would.